core/src/main/java/com/scalar/db/transaction/consensuscommit/TransactionalTable.java

```diff
@@ -0,0 +1,42 @@
+package com.scalar.db.transaction.consensuscommit;
+
+import com.google.common.collect.ImmutableMap;
+import com.scalar.db.api.TableMetadata;
+import com.scalar.db.io.DataType;
+
+public final class TransactionalTable {
+
+  private static final ImmutableMap<String, DataType> TRANSACTION_META_COLUMNS =
+      ImmutableMap.<String, DataType>builder()
+          .put(Attribute.ID, DataType.TEXT)
+          .put(Attribute.STATE, DataType.INT)
+          .put(Attribute.VERSION, DataType.INT)
+          .put(Attribute.PREPARED_AT, DataType.BIGINT)
+          .put(Attribute.COMMITTED_AT, DataType.BIGINT)
+          .put(Attribute.BEFORE_ID, DataType.TEXT)
+          .put(Attribute.BEFORE_STATE, DataType.INT)
+          .put(Attribute.BEFORE_VERSION, DataType.INT)
+          .put(Attribute.BEFORE_PREPARED_AT, DataType.BIGINT)
+          .put(Attribute.BEFORE_COMMITTED_AT, DataType.BIGINT)
+          .build();
+
+  private TransactionalTable() {}
+
+  /**
+   * Converts a table metadata to a transactional one
+   *
+   * @param tableMetadata a table metadata to be converted
+   * @return a transactional table metadata
+   */
+  public static TableMetadata convertToTransactionalTable(TableMetadata tableMetadata) {
```

Comment:
Should `TableMetadata` have a `transactional` boolean flag or something to make it safer even if the same tablemetadata object is applied to this method multiple times? (it would add `before_before_column1` or something)
It is not gonna happen probably, but I just feel it might be safer and it might be useful if we can  easily check a given tablemetadata object is transactional or not. (instead of checking if there is a `before_` column)
What do you think?

Reply:
The transactional table things are `Consensus Commit` implementation specific things, so I don't think `TableMetadata` should have a `transactional` boolean flag or something like that.

I'm wondering if we really need to take care of the situation where the same `TableMetadata` object is applied to the method multiple times. Or is there any situation we need to check if a `TableMetadata` is transactional or not? If we need to do that, I think we need to check if a `TableMetadata` has transaction meta columns or not.

Reply:
I added a check if a TableMetadata has transaction meta columns or not in 5e2f7d9d77496a74e74a6bfed6ef051980986be8. Please take a look!